Eyevinn/osaas-client-ts
# feat(cli): add --version option to display CLI version (#18)

# Add Version Option to CLI

This PR implements a new feature allowing users to check the version of the CLI tool using the `--version` or `-v` flag.

## Changes

- Added functionality to read the version from `package.json`
- Implemented the version option using Commander.js in `cli.ts`
- The CLI now responds to `osc --version` and `osc -v` commands

## Implementation Details

- Used `fs.readFileSync` to read the `package.json` file
- Utilized the `path.join` method to ensure cross-platform compatibility
- Integrated the version information with Commander.js's `version` method

## Testing

To test this feature:

1. Build the project: `npm run build`
2. Install the CLI globally: `cd packages/cli && npm install -g .`
3. Run `osc --version` or `osc -v` from any directory

Expected output: The current version number (e.g., `0.10.1`)

## Additional Notes

- This implementation assumes the CLI package's `package.json` is one level up from the `cli.ts` file
- The global installation step is for testing purposes and is not part of the committed code

Closes #18
